src/main/java/org/openrewrite/apache/httpclient5/MigrateRequestConfig.java

```diff
@@ -0,0 +1,161 @@
+    private static class MigrateRequestConfigPrecondition extends JavaIsoVisitor<ExecutionContext> {
+
+        @Override
+        public J.MethodInvocation visitMethodInvocation(J.MethodInvocation method, ExecutionContext ctx) {
+            if (MATCHER_STALE_CHECK_ENABLED.matches(method) && !Boolean.parseBoolean(method.getArguments().get(0).print())) {
+                return SearchResult.found(method);
+            }
+            return super.visitMethodInvocation(method, ctx);
+        }
+    }
+
+    private static class MigrateRequestConfigVisitor extends JavaIsoVisitor<ExecutionContext> {
+
+        @Override
+        public J.MethodInvocation visitMethodInvocation(J.MethodInvocation method, ExecutionContext ctx) {
+            method = super.visitMethodInvocation(method, ctx);
+
+            if (MATCHER_STALE_CHECK_ENABLED.matches(method)) {
+                boolean enabled = Boolean.parseBoolean(method.getArguments().get(0).print());
+                getCursor().putMessageOnFirstEnclosing(J.MethodDeclaration.class, KEY_STALE_CHECK_ENABLED, enabled);
+                getCursor().putMessageOnFirstEnclosing(J.MethodDeclaration.class, KEY_REQUEST_CONFIG, method);
+                doAfterVisit(new RemoveMethodInvocationsVisitor(Collections.singletonList(PATTERN_STALE_CHECK_ENABLED)));
+            } else if (MATCHER_REQUEST_CONFIG.matches(method)) {
+                Set<Tree> connManagers = new HashSet<>(TreeVisitor.collect(
+                        new JavaIsoVisitor<ExecutionContext>() {
+                            @Override
+                            public J.VariableDeclarations visitVariableDeclarations(J.VariableDeclarations multiVariable, ExecutionContext ctx) {
+                                if (TypeUtils.isOfClassType(multiVariable.getTypeAsFullyQualified(), FQN_POOL_CONN_MANAGER)) {
+                                    return SearchResult.found(multiVariable);
+                                }
+                                return super.visitVariableDeclarations(multiVariable, ctx);
+                            }
+                        },
+                        getCursor().firstEnclosing(J.MethodDeclaration.class),
+                        new HashSet<>()
+                ));
+
+                // Call `setConnectionManager()` if there's no PoolingHttpClientConnectionManager
+                // The `poolingHttpClientConnectionManager` will be created later in `visitMethodDeclaration()`
+                if (connManagers.isEmpty()) {
+                    method = JavaTemplate.builder(method.print() + "\n.setConnectionManager(poolingHttpClientConnectionManager);").
+                            javaParser(JavaParser.fromJavaVersion().classpath(JavaParser.runtimeClasspath()))
+                            .imports("org.apache.hc.client5.http.impl.io.PoolingHttpClientConnectionManager")
+                            .build()
+                            .apply(updateCursor(method), method.getCoordinates().replace());
```

Comment:
The recommended pattern here is something like
```suggestion
                    method = JavaTemplate.builder("#{any()}.setConnectionManager(poolingHttpClientConnectionManager);").
                            javaParser(JavaParser.fromJavaVersion().classpath(JavaParser.runtimeClasspath()))
                            .imports("org.apache.hc.client5.http.impl.io.PoolingHttpClientConnectionManager")
                            .build()
                            .apply(updateCursor(method), method.getCoordinates().replace(), method);
```

Reply:
I would like to aggregate both JavaTemplate to `visitMethodDeclaration`, the logic maybe clearer. I tried but seems not possible.